### ISSUE TITLE
Feel confusing about the cluster loss and separation loss

### ISSUE THREAD

**Issue by qzzdd** [zaixizhang/ProtGNN#1]
In the paper, the cluster loss is defined as the average minimum of the distance between each graph embedding and the prototype of its own class. However, in the code:
```python
cluster_cost = torch.mean(torch.min(min_distances * prototypes_of_correct_class, dim=1)[0])
```
The multiplication "min_distances * prototypes_of_correct_class" enables the "distance" of the other class equal to zero, thus the minimum is zero instead of the minimum corresponding distance. As a result,  the cluster cost is always zero during the training process, and it doesn't work as expected.

The separation loss has the same problem.

Looking forward to your reply. Thanks!

**Comment by zaixizhang** [comment]
Hi,
Thanks for your question! We have updated our code to mask the zero elements.
